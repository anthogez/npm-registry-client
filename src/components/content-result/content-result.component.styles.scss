.content-result {
	margin-top: 2.5%;
}
