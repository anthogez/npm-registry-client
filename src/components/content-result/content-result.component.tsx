import React from 'react';
import { ErrorMessage, NodeTreeView, LoadingRequest } from '..';

import './content-result.component.styles.scss';
import config from '../../config';
const { loadingRequestProps } = config.components;

const ContentResult: React.FC<any> = ({
	errorMessage,
	nodePackage,
	isLoading
}) => {
	return (
		<div>
			{isLoading ? <LoadingRequest {...loadingRequestProps} /> : null}
			{!isLoading && errorMessage ? (
				<ErrorMessage errorMessage={errorMessage} />
			) : null}
			{nodePackage ? <NodeTreeView nodePackage={nodePackage} /> : null}
		</div>
	);
};

export default ContentResult;
