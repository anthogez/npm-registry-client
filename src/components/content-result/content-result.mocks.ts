export const mockedNodePackage = {
	name: 'minimatch',
	version: '3.0.4',
	dependencies: [
		{
			name: 'brace-expansion',
			version: '1.1.7',
			dependencies: [
				{
					name: 'balanced-match',
					version: '0.4.1',
					dependencies: []
				},
				{
					name: 'concat-map',
					version: '0.0.1'
				}
			]
		}
	]
};