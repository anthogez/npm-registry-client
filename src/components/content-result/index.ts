import ContentResult from './content-result.component';
export default ContentResult;
