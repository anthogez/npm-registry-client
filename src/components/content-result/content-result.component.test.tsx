import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import ContentResult from './';
import { mockedNodePackage } from './content-result.mocks';

configure({ adapter: new Adapter() });

it('Test if ContentResult Component renders without crash', () => {
	const contentResult = shallow(
		<ContentResult
			errorMessage="RIP, Not Found :)"
			nodePackage={null}
			isLoading={false}
		/>
	);
	expect(contentResult).toMatchSnapshot();
});

describe('This is a group of inner tests of ContentResult Component', () => {
	it('Test if on set isLoading as <false> and assigning a value to errorMessage props, ErrorMessage component will be displayed', () => {
		const contentResult = shallow(
			<ContentResult
				errorMessage="RIP, Not Found :)"
				nodePackage={null}
				isLoading={false}
			/>
		);
		expect(contentResult.find('ErrorMessage')).toHaveLength(1);
	});

	it('Test if on set isLoading <true> LoadingRequest Component will be rendered', () => {
		const contentResult = shallow(
			<ContentResult errorMessage="" nodePackage={null} isLoading={true} />
		);
		expect(contentResult.find('LoadingRequest')).toHaveLength(1);
	});

	it('Test if on set isLoading <false> LoadingRequest Component will be rendered', () => {
		const contentResult = shallow(
			<ContentResult errorMessage="" nodePackage={null} isLoading={false} />
		);
		expect(contentResult.find('LoadingRequest')).toHaveLength(0);
	});

	it('Test if on set isLoading <false>, LoadingRequest Component will be rendered', () => {
		const contentResult = shallow(
			<ContentResult errorMessage="" nodePackage={{}} isLoading={false} />
		);
		expect(contentResult.find('LoadingRequest')).toHaveLength(0);
	});

	it('Test if on set a mocked object <mockedNodePackage> on nodePackage props, NodeTreeView Component will be rendered', () => {
		const contentResult = shallow(
			<ContentResult
				errorMessage=""
				nodePackage={mockedNodePackage}
				isLoading={false}
			/>
		);
		expect(contentResult.find('NodeTreeView')).toHaveLength(1);
	});
});
