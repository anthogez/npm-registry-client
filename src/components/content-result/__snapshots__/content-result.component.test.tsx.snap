// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if ContentResult Component renders without crash 1`] = `ShallowWrapper {}`;
